engine/runtime/src/main/scala/org/enso/compiler/Compiler.scala

```diff
@@ -1016,10 +998,68 @@ class Compiler(
     diagnostic: IR.Diagnostic,
     source: Source
   ): String = {
-    fileLocationFromSection(
-      diagnostic.location,
-      source
-    ) + ": " + diagnostic.formattedMessage
+    val outSupportsAnsiColors = System.console() != null
+    val ansiReset             = "\u001B[0m"
```

Comment:
There is a library doing all the fancy coloring already: https://github.com/com-lihaoyi/fansi
Maybe you want to check it out or get some inspiration

Reply:
I'd also recommend considering using this library, maybe it could solve the issue with the ANSI escapes rendering badly in `conhost` on Windows?

Reply:
The newest version of the PR already uses `fansi` library. But still, the only recommended way how to check if an output supports ANSI coloring is via `System.getConsole() != null`.